pkg/cache/cluster_info.go

```diff
@@ -265,6 +266,7 @@ func (m *ClusterInfo) processNodeUpdate(request *si.UpdateRequest) {
                         zap.String("nodeId", node.NodeId),
                         zap.String("partition", nodeInfo.Partition))
                     acceptedNodes = append(acceptedNodes, &si.AcceptedNode{NodeId: node.NodeId})
+                    existingAllocations = append(existingAllocations, node.ExistingAllocations...)
```

Comment:
I have one question which I can not find in the changes: how does this interact with the node registration? When a node registers it also provides a list of existing allocations which get added to the PartitionInfo in the cache when we call partition.addNewNode 6 lines above here through this:
'''
    // Add allocations that exist on the node when added
    if len(existingAllocations) > 0 {
        log.Logger.Info("add existing allocations", ...
  ...
'''
Since we are replaying the allocation already should we still have this?
How do we handle a node re-registration outside of the shim recovery which is what we do now as far as I am aware.

Reply:
The code you pasted upon was the place where we handle existing allocations in the cluster-level cache, e.g `PartitionInfo`, `QueueInfo`. The reason to collect `existingAllocations` here is to send this info to the scheduler so the scheduler can update its cache, aka `SchedulingQueue`, `SchedulingApplication` etc.

Reply:
For node re-registration, I don't think we allow that.
```
 if node := pi.nodes[node.NodeId]; node != nil {
        return fmt.Errorf("partition %s has an existing node %s, node name must be unique", pi.Name, node.NodeId)
    }
```
if shim sends 2 same node as NewNodeInfo in UpdateRequest, the 2nd one will be rejected.

Reply:
My concern is that we can have a loop back: update in one place (cache or scheduler) which generates an event to update the other side which we already did update. I am not afraid about the same node registering twice we have that case covered.

As long as we're sure that there is no feedback loop or we handle it correctly by breaking the loop then I am OK. Just not 100% sure that we have coded that defensively.

Reply:
Based on off-line conversations there should not be a loop as it breaks on the fact that an ID can only be added once and the loop will stop.